### ISSUE TITLE
[BUG] Stuck alerts in Netdata Cloud for agents that are no longer live.

### ISSUE THREAD

**Issue by dimko** [netdata/netdata-cloud#217]
Stuck alarms happen when an agent/node has an alarm raised on its behalf and the agent/node that raised it goes offline.

![image](https://user-images.githubusercontent.com/19268853/145553160-66a1d335-40b1-4909-8e0e-dfee418c100f.png)

This mostly affects agents with parent/child setup e.g Parent has the alert configuration for a child agent, so he is the one raising the alert for that child. If the Parent dies, the alert for the child is not cleared.

- [X] https://github.com/netdata/cloud-schemas/pull/91
- [X] https://github.com/netdata/cloud-alarm-streaming-service/pull/22
- [X] https://github.com/netdata/cloud-alarm-processor-service/pull/34
- [X] https://github.com/netdata/cloud-nodes-service/pull/345
- [X] https://github.com/netdata/cloud-alarm-notification-sender-service/pull/30
- [X] https://github.com/netdata/cloud-alarm-processor-service/pull/35
- [X] https://github.com/netdata/product/issues/2653
- [X] https://github.com/netdata/netdata/pull/11960
- [X] https://github.com/netdata/cloud-alarm-streaming-service/pull/23
- [X] https://github.com/netdata/netdata/pull/11965
- [X] https://github.com/netdata/netdata/pull/12021
- [X] https://github.com/netdata/cloud-alarm-processor-service/pull/46
- [X] https://github.com/netdata/cloud-alarm-processor-service/pull/47

**Comment by dimko** [comment]
@car12o Is this something you've implemented and is pending review? If yes, can you please link the PR for it? If not, I can't understand why it is in the Under PR section. 

cc @manos-saratsis

**Comment by car12o** [comment]
> @car12o Is this something you've implemented and is pending review? If yes, can you please link the PR for it? If not, I can't understand why it is in the Under PR section.
> 
> cc @manos-saratsis

Linked the relative PR's ... thanks & sorry for missing it :wink:

**Comment by dimko** [comment]
This was fixed as part of the following releases:

/release cloud-nodes-service release/1.48.0
/release cloud-alarm-processor-service release/0.3.1
/release cloud-alarm-notification-sender-service release/0.4.0
/release cloud-alarm-streaming-service release/0.4.0
/release cloud-alarm-processor-service release/0.3.2

@car12o I am closing this. Great job!!

(will re-open if we need anything else)

**Comment by dimko** [comment]
Agent issue is fixed on this PR: https://github.com/netdata/netdata/pull/11928

Moving to acceptance stage

**Comment by dimko** [comment]
Moving back to In progress since an extra fix from agent side is required.

**Comment by dimko** [comment]
Agent PR to fix agent health issue causing issues regarding the stucked alerts https://github.com/netdata/netdata/pull/11960

**Comment by car12o** [comment]
The cloud PR to address this issue regarding the stuck alarms.
https://github.com/netdata/cloud-alarm-streaming-service/pull/23

**Comment by stelfrag** [comment]
Handle child health config state after reconnection when health is set to `auto` netdata/netdata#11965

**Comment by ralphm** [comment]
Related Agent PR: netdata/netdata#12021

**Comment by car12o** [comment]
All the fixes have been deployed to production today!

**Comment by phiberoptick** [comment]
Sorry to chime in on a closed issue but I found this while trying to find a way to clear defunct alarms.  I see there hasn't been any activity in quite awhile but it appears that I am having the same issue.  As far as the agents are concerned the alarms don't exist.  Trying to click and follow them in the cloud interface leads to a page stating that chart/metric doesn't exist (although the metrics do exist and are continuing to update.

In my case, I have suddenly (over the past couple of weeks) been getting constant floods of anomaly alarms that will almost immediately clear.  They vary as to the metric they are alarming on.  Usually ip, ram, hardirq etc.

The problem is that at one point while there were 60 or so alarms in the cloud, I was performing maintenance on all of the children (15 or so) as well as the parent.  During this time, all the agents were updated and the servers were restarted.

The restarts were for resource increases and I think that briefly there was high RAM usage and Netdata may have been killed...unfortunately this may have also happened while it was temporarily unable to write to the disk as well.  The reboots should've been clean and it appeared that way.  Netdata seemed to start normally on the parent and children but ever since they came back up the cloud interface is still persisting the alerts (albeit with no actual values listed for the metric in the alerts tab list) while the children and parent nodes show no alarms and seem to be otherwise running normally...

Is there any news on this or way to clear the alarms out of the cloud interface?  Restarting the agents doesn't seem to do anything except change the "Triggered" time to when the agent was restarted.

![screencapture-app-netdata-cloud-spaces-rooms-all-nodes-alerts-2022-08-14-08_24_02-nodes_redacted](https://user-images.githubusercontent.com/2384869/184536741-72b42f16-98cb-4339-ba91-8939e5cf3382.png)
